tests/unit/IonTextReaderTest.js

```diff
@@ -125,6 +125,39 @@
       assert.equal(ionReader.fieldName(), "innerkey2");
     };
 
+      suite['Accept IVM like symbols and throw on IVMs except $ion_1_0'] = function() {
```

Comment:
the nested try/catch structure is unusual, and there's quite a bit of repetition.  i'd suggest defining a helper function with parameters for the string to test and an indication of what the expected behavior is (false/true/error).

and let's add `$ion_1`, `$ion_1_1`, and `$ion_1_a` as additional test cases